src/modules/ims_ipsec_pcscf/cmd.c

```diff
@@ -689,8 +652,13 @@ int ipsec_create(struct sip_msg* m, udomain_t* d, int _cflags)
 
     struct sip_msg* req = t->uas.request;
 
+
+        security_t* req_sec_params = NULL;
+        req_sec_params = cscf_get_security(req);
+
+
     // Update contacts only for initial registration, for re-registration the existing contacts shouldn't be updated.
-    if(ci.via_port == SIP_PORT){
+    if(ci.via_port == SIP_PORT && req_sec_params == NULL){
```

Comment:
**ipsec_create()** should be invoked exactly before 401 reply. From my practice in received REGISTER on 5060 there is always **sec-agree** header and this **if** block will never been TRUE.